Formula/q/qurancli.rb

```diff
@@ -0,0 +1,20 @@
+class Qurancli < Formula
+  desc "Simple tool to Read, Search and Recite the Quran"
+  homepage "https://github.com/ahmedsaheed/quranCLI"
+  url "https://github.com/ahmedsaheed/quranCLI/releases/download/v1.0.0/qurancli-1.0.zip"
+  version "1.0.0"
+  sha256 "572b1d3073e318876a95e1fa38cd539cae8490336a6c6f4c9ce20f6887cbc042"
+  license "Apache-2.0"
+
+  depends_on "openjdk@17"
+
+  def install
+    libexec.install Dir["*"]
+    bin.install_symlink "#{libexec}/bin/qurancli" => "qurancli"
+  end
+
+  test do
+    output = shell_output("#{bin}/qurancli --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.